### DISCUSSION TITLE
Serving files from web server

### DISCUSSION THREAD

**Discussion by debjanipaul** [h2oai/wave#1545] [Q&A]
I was able to follow the tutorial https://wave.h2o.ai/docs/files#serving-files-directly-from-the-wave-server and serve a file and access it from from my localhost. I want to know how do I do the same from the H2O app store. Do I have to configure the toml file and what parameters should I use for 'waved -public-dir /datasets/@/home/zaphod/data -public-dir /public/models/@/home/zaphod/models' in the toml file to replicate the same behavior?

**Comment by mturoci** [comment]
Paths need to be relative to Wave server binary location.

## Solution

The following conf makes `assets` folder to be served from Wave server in app store:

```toml
# app.toml

[[Env]]
Name = "H2O_WAVE_PUBLIC_DIR"
Value = "/assets/@../../app/static/"
```

Note: The `assets` folder is assumed to reside in your app root.

## Explanation

The Wave server is located at `../../app` relative to the current working directory (where your app gets extracted) in app store.

A good candidate for docs update cc @mtanco.

**Reply by debjanipaul** [reply]
HI @mturoci, I am launching the local web server as:
```
 ./waved -public-dir /datasets/@/Users/<userID>/Documents/GitRepos/visualization_/geo_data
```
The geo_data folder is in the app root and has the geojson file. To point plotly to access the geojson file I am including the following line of code in my .py file: 
```
counties = 'http://localhost:10101/datasets/geojson-counties-fips-mod.json
```

Do I need to have the geojson data inside assets folder or the current geo_data folder will work?
```
# Will this work in the app.toml

[[Env]]
Name = "H2O_WAVE_PUBLIC_DIR"
Value = "/geo_data/@../../app/static/"

```

To access the geo_json data in .py file, will this work?
```
counties = 'http://companyURL.com/geo_data/geojson-counties-fips-mod.json
```

**Reply by mturoci** [reply]
It's hard to guess your project structure, but it seems you need to try using `/geo_data/@../../app/geo_data/` as a value (if the geodata folder is in the root of your project dir).

If you wish, you can also put the `geo_data` into a `static` folder and use the code you suggested in the previous comment.
